src/util/ChartUtils.ts

```diff
@@ -45,25 +46,31 @@ export function getValueByDataKey<T>(obj: T, dataKey: DataKey<T>, defaultValue?:
   return defaultValue;
 }
 /**
- * Get domain of data by key
+ * Get domain of data by key.
  * @param  {Array}   data      The data displayed in the chart
  * @param  {String}  key       The unique key of a group of data
  * @param  {String}  type      The type of axis
  * @param  {Boolean} filterNil Whether or not filter nil values
  * @return {Array} Domain of data
  */
-export function getDomainOfDataByKey<T>(data: Array<T>, key: string, type: string, filterNil?: boolean) {
-  const flattenData = _.flatMap(data, entry => getValueByDataKey(entry, key));
+export function getDomainOfDataByKey<T>(
+  data: Array<T>,
+  key: DataKey<T>,
+  type: BaseAxisProps['type'],
+  filterNil?: boolean,
+): NumberDomain | CategoricalDomain {
+  const flattenData: unknown[] = _.flatMap(data, (entry: T): unknown => getValueByDataKey(entry, key));
 
   if (type === 'number') {
-    const domain = flattenData.filter(entry => isNumber(entry) || parseFloat(entry));
+    // @ts-expect-error parseFloat type only accepts strings
+    const domain: number[] = flattenData.filter(entry => isNumber(entry) || parseFloat(entry));
```

Comment:
Nit: Naming
This is not the domain yet, it is only the parsedData, right?

Reply:
Maybe, I didn't add the variable, only the @ts- comment